out_oracle_log_analytics: Implement IMDS authentication, chunking

rghouzra opened the pull request:
## Problem
Oracle cloud infrastructure users running Fluent Bit on oci compute instances need a way to authenticate and send logs to oracle log analytics without managing api keys manually. Instance principal authentication provides a secure, automated way to authenticate using the instance's identity.

## Change
This PR adds oci Instance Principal authentication support to the `out_oracle_log_analytics` plugin:

- **IMDS Authentication**: Implemented Oracle Instance Metadata Service authentication with automatic security token refresh, certificate, and key retrieval from instance metadata
- **Log Chunking**: Implemented log chunking and batching with size estimation for handling large payloads
- **Timezone Support**: Added timezone configuration for accurate timestamp
- **enhanced metadata extraction**: Improved metadata extraction with dot-notation support

### Key Configuration Options Added:
- `auth_type`: Choose between `config_file` or `instance_principal`auths, also by if the user didn't specify it, will be by default config_file> 
- `domain_suffix`: Optional OCI domain suffix for different realms
- `oci_la_timezone`: Timezone configuration for log timestamps (the supported timzeones by oci log analytics)
- `dump_payload_file` / `payload_file_location`: Debug options to dump payloads locally

## Testing
- [x] Unit tests added covering instance principal auth like imds failure scenarios, and config-file authentication
- [x] Example configuration files provided (see below)
- [ ] Attached Valgrind output that shows no leaks or memory corruption was found
- [ ] Debug log output from testing the change
- [N/A] Run local packaging test showing all targets build
- [N/A] Set `ok-package-test` label to test for all targets

### example configuration file:

**Instance Principal Authentication:**
```ini
[OUTPUT]
    Name oracle_log_analytics
    
    # new: Instance principal auth by default its config auth
    auth_type instance_principal
    
    # this one is optional timezone configuration
    oci_la_timezone UTC
    
    # Required oci Log analytics   parameters
    namespace your_tenancy_namespace
    oci_la_log_group_id ocid1.loggroup.oc1..example
    oci_la_log_source_name YourLogSource

edsiper commented on plugins/out_oracle_log_analytics/oci_logan.c:
```diff
@@ -620,16 +658,372 @@ static flb_sds_t compose_uri(struct flb_oci_logan *ctx,
     return full_uri;
 }
 
+// Checks if the current security token needs refresh(expires within 5min)
```
use /* */ for single line https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#single-line-comment

edsiper commented on plugins/out_oracle_log_analytics/oci_logan.c:
```diff
@@ -620,16 +658,372 @@ static flb_sds_t compose_uri(struct flb_oci_logan *ctx,
+        return 0;
+    }
+
+    flb_sds_t json_payload = create_federation_payload(ctx);
```
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#variable-definitions

edsiper commented on plugins/out_oracle_log_analytics/oci_logan.c:
```diff
@@ -620,16 +658,372 @@ static flb_sds_t compose_uri(struct flb_oci_logan *ctx,
+    }
+
+    flb_sds_t response = sign_and_send_federation_request(ctx, json_payload);
+    flb_sds_destroy(json_payload);
```
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#variable-definitions

edsiper commented on plugins/out_oracle_log_analytics/oci_logan.c:
```diff
@@ -1005,200 +1584,403 @@ static int get_and_pack_oci_fields_from_record(msgpack_packer *packer,
+            (remaining_records >
+             records_per_batch) ? records_per_batch : remaining_records;
+
+        ret =
```
complete the line here

rghouzra replied:
i

> complete the line here

thats done by the gnu indent with apache format style `If an expression (or a routine declaration or invocation) would extend past column 80, the terms or arguments are wrapped at a convenient spot and the wrapped portion is indented under the first term in the expression (or the first argument to the function).`

edsiper reviewed (changes requested):
please adjust the code style https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#coding-style (not only for my comments but in general)

rghouzra commented:
Any update on this

rghouzra committed:
out_oracle_log_analytics: add IMDS authentication and chunking

Adds Instance Metadata Service based authentication support with
federation token signing, JWT handling, log chunking for large
payloads, and timezone support.

Signed-off-by: rghouzra <rghouzra@student.1337.ma>
Signed-off-by: dexter <rghouzra@student.1337.ma>

rghouzra committed:
tests: add oracle log analytics IMDS runtime tests

Signed-off-by: rghouzra <rghouzra@student.1337.ma>
Signed-off-by: dexter <rghouzra@student.1337.ma>

rghouzra committed:
Merge branch 'master' into prod_v1.0.2

Signed-off-by: Reda Ghouzraf <rghouzra@student.1337.ma>

rghouzra committed:
out_oracle_log_analytics: fix incomplete merge of IMDS chunking logic

Signed-off-by: Reda Ghouzraf <reda.ghouzraf@oracle.com>

rghouzra commented:
@patrick-stephens @edsiper any update on this please
